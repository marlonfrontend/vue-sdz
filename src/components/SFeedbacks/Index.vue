<template>
  <coe-feedback :feedbacks="feedbacks" v-on="$listeners">
    <template slot="feedbacks" slot-scope="{ feedbacks, success, error }">
      <s-card
        v-for="(feedback, index) in feedbacks"
        :key="index"
        :class="getClasses(feedback.type)"
      >
        <div class="message-container">
          <span class="message">{{ feedback.message }} </span>

          <span :style="{ color: getColor(feedback.type, success, error) }" class="highlighted">
            {{ feedback.highlighted }}
          </span>
        </div>

        <s-button
          v-if="dismissible"
          grey
          icon="plus"
          class="button"
          @click="$emit('close', feedback)"
        />
      </s-card>
    </template>
  </coe-feedback>
</template>

<script>
import SCard from '../SCard/Index.vue'
import SButton from '../SButton/Index.vue'
import { CoeFeedback } from 'vue-coe-feedback'

export default {
  name: 'SFeedbacks',

  components: { CoeFeedback, SCard, SButton },

  props: {
    feedbacks: {
      type: Array,
      default: () => []
    },

    dismissible: {
      type: Boolean,
      default: true
    }
  },

  methods: {
    getClasses (type) {
      return ['feedback', {
        '--is-negative': type === 'error',
        '--is-positive': type === 'success'
      }]
    },

    getColor (type, success, error) {
      const colors = { success, error }

      return colors[type]
    }
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.coe-feedbacks {
  & > .feedback {
    margin-left: 20px;
    min-height: 60px;
    box-shadow: -1px -1px 4px 0 rgba(0,0,0,0.2);

    &.--is-negative {
      background-color: color(negative, dark) !important;

      & > .message-container > .message, .highlighted { color: $neutral-color !important; }
    }

    &.--is-positive {
      background-color: color(primary, light) !important;

      & > .message-container > .message, .highlighted  { color: $neutral-color !important; }
    }

    &:last-child { margin-bottom: 20px; }

    & > .message-container {
      display: flex;
      align-items: center;
      flex-direction: column;

      font-weight: unset;
      color: color(primary, base) !important;

      & > .message { font-weight: $font-weight-bold; }
      & > .highlighted { font-weight: $font-weight-regular; }
    }

    // override s-button
    & > .button {
      width: 30px;
      margin-right: 15px;
      height: 30px !important;
      min-width: unset !important;
      min-height: unset !important;
      border-radius: 50% !important;
      border: 1px solid rgba(#FFF, .3);

      & > .icon {
        fill: rgba(#FFF, .3);
        margin: 0 !important;
        width: 14px !important;
        height: 14px !important;
      }

      &:hover {
        border-color: rgba(#FFF, .8);
        & > .icon { fill: rgba(#FFF, .8) }
      }
    }
  }
}
</style>
