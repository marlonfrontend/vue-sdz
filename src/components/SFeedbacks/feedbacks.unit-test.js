import { mount } from '@vue/test-utils'

import SFeedbacks from './Index.vue'

/* eslint-disable */
describe('SFeedbacks', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SFeedbacks, {
      propsData: {
        feedbacks: [
          { type: 'info', message: 'teste' },
          { type: 'info', message: 'Nova informação' }
        ]
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })

  test('check count messages', () => {

    const text = cmp.findAll('.message-container')
    expect(text.length).toBe(2);

  })

  test('check body text', () => {

    const text = cmp.find('is-positive')

    expect(text.exists()).toBe(false)

  })

})
