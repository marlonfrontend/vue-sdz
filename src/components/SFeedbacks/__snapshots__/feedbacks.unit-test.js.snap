// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SFeedbacks has the expected html structure 1`] = `
<transition-group-stub
  class="coe-feedbacks"
  name="feedback"
  tag="ul"
>
  <div
    class="s-card feedback"
    style="background: white;"
  >
    <div
      class="message-container"
    >
      <span
        class="message"
      >
        teste 
      </span>
       
      <span
        class="highlighted"
      >
        
          
        
      </span>
    </div>
     
    <button
      class="c-link button s-button --grey --icon-only"
    >
      <i
        class="icon s-icon icon sdz-close"
        style="font-size: 20px;"
      />
       
      <!---->
       
      <transition-stub
        name="loading"
      >
        <!---->
      </transition-stub>
    </button>
  </div>
  <div
    class="s-card feedback"
    style="background: white;"
  >
    <div
      class="message-container"
    >
      <span
        class="message"
      >
        Nova informação 
      </span>
       
      <span
        class="highlighted"
      >
        
          
        
      </span>
    </div>
     
    <button
      class="c-link button s-button --grey --icon-only"
    >
      <i
        class="icon s-icon icon sdz-close"
        style="font-size: 20px;"
      />
       
      <!---->
       
      <transition-stub
        name="loading"
      >
        <!---->
      </transition-stub>
    </button>
  </div>
</transition-group-stub>
`;
